### ISSUE TITLE
Allow changing AppStream IDs

### ISSUE THREAD

**Issue by Pointedstick** [flathub/flathub#227]
Right now Flathub doesn't have a facility to rename AppStream IDs (perhapos because they're used as the repo names?). This is problematic in cases where the AppStream ID used by Flathub is wrong and we would like to change it so that we can match up with upstream data.

Examples are:
- Endless Sky: https://github.com/flathub/io.github.EndlessSky.endless-sky/issues/2
- FileZilla: https://github.com/flathub/org.filezillaproject.Filezilla/issues/2

**Comment by TingPing** [comment]
> (perhapos because they're used as the repo names?)

Renaming a repo on github is easy. The OSTree repo they get exported to needs to figure out redirecting old installs to the new one.

**Comment by TingPing** [comment]
Looks like `io.elementary.code` should be renamed to `org.elementary.code`

**Comment by Alexander-Wilms** [comment]
`io.elementary.code` is the ID they use: https://github.com/elementary/code/tree/master/data

**Comment by TingPing** [comment]
@Alexander-Wilms oh well this should be fixed upstream then?
 https://github.com/flathub/io.elementary.code/blob/master/io.elementary.code.json#L13-L14

**Comment by Alexander-Wilms** [comment]
No, these should be io.elementary.[...] as well. I fixed it now.

**Comment by TingPing** [comment]
`org.videolan.VLC` -> `org.videolan.vlc` is another.

**Comment by joergister** [comment]
Also 
`org.audacityteam. Audacity` > `org.audacityteam. audacity`

**Comment by gpoo** [comment]
We are evaluating to rename org.gnome.Evince -> org.gnome.evince.

The rationale is that we have two D-Bus interfaces (application and daemon), both with org.gnome.evince. Flatpak's org.gnome.evince.Daemon cannot be launched it cannot be owned by org.gnome.Evince.

Alternatively, we can break the D-Bus ABI (and/or try to support both org.gnome.Evince and org.gnome.evince D-Bus interfaces). If renaming the app-id were not a major issue, we are more likely to go for it.

**Comment by ramcq** [comment]
@gpoo You don't need to change the name for this - if you define org.gnome.evince.Daemon=own in the Flatpak session bus policy, you can export this name and a service activation file.

**Comment by gpoo** [comment]
@ramcq thanks for the tip. I will give it a try. That would give us a big relief :-)

**Comment by TingPing** [comment]
It still makes sense in the long term to have them under the same namespace.

**Comment by gpoo** [comment]
@TingPing ok. However, we can delay the name change until we need to break the D-Bus ABI, so we can break it only once. We have some candidates, but not the time to do it within the next one or two release cycles. Thus, owning the service alleviates the situation.

**Comment by mcatanzaro** [comment]
> We are evaluating to rename org.gnome.Evince -> org.gnome.evince.

We're trying to standardize GNOME on the former (org.gnome.Evince).

**Comment by gpoo** [comment]
I had forgotten the issue after we set:

```
   "finish-args": [
        ...
        "--own-name=org.gnome.evince",
        "--own-name=org.gnome.evince.Daemon"
    ],
```

As I remember from discussions with other Evince developers, it should stay as it is unless there is a good reason to break the D-Bus ABI.

**Comment by Zlopez** [comment]
I'm trying to update `gnome-chess` to latest version and I'm hitting issue with D-Bus (See https://github.com/flathub/org.gnome.chess/pull/8), which is probably caused by the name.

My question is: "Is the only way to do a rename right now to deprecate the existing flatpak and create a new one?"
Or did the situation changed?

**Comment by gpoo** [comment]
@Zlopez the solution is in the comment right before yours. If you read the discussion, you should get the context as well.

**Comment by Zlopez** [comment]
@gpoo 
I tried your solution, but the test build failed. See https://flathub.org/builds/#/builders/32/builds/2210
Did I missed something or do we have different issue?

**Comment by ramcq** [comment]
@Zlopez why are you changing from the good situation (where the app ID matches the D-Bus name) to the bad one (where this is not true any more)?

**Comment by ramcq** [comment]
> I tried your solution, but the test build failed. See https://flathub.org/builds/#/builders/32/builds/2210

This build failed because Flathub is still building the app ID org.gnome.chess and expecting that it has that app ID. So now the ID inside the manifest doesn't match the ID that Flathub expects. You would need a new repo and a new manifest named org.gnome.Chess to rebuild a new app with this name.

But really, why this madness? Once you have the ID, all you do by changing it is break loads of stuff - if people have the desktop icon pinned, and you rename the app ID and desktop file, you leave them a dead icon. Never mind Flatpak, but Shell also doesn't have a way to support renaming apps. What's the goal?

**Comment by Zlopez** [comment]
@ramcq 
I asked in my previous post if I should deprecate this repository and create a new one and @gpoo pointed me to his solution, so I tried it.

**Comment by bilelmoussaoui** [comment]
As Flathub can finally rename app-ids, here's a list of issues that would be nice to get them finally fixed
- [ ] https://github.com/flathub/net.minetest.Minetest/issues/5
- [ ] https://github.com/flathub/org.mozilla.Thunderbird/issues/59
- [ ] https://github.com/flathub/org.zulip.Zulip/issues/4

**Comment by Zlopez** [comment]
@bilelmoussaoui Isn't Thunderbird already EOL?

**Comment by aral** [comment]
Hiya, I need to rename ind.ie.Gnomit to org.small-tech.Gnomit. [I’ve been told](https://github.com/flathub/linux-store-frontend/issues/218#issuecomment-617465614) that I should open a new merge request here and mark the old repo as end of life.

Given @bilelmoussaoui’s commend (“as Flathub can finally rename app-ids”), I wanted to see if that’s still the recommended way forward or is there a newer way.

Thanks in advance!

**Comment by bilelmoussaoui** [comment]
As mentioned before, just open a new merge request with the new app-id and one of the reviewers will merge it and end of life the old app-id.

**Comment by aral** [comment]
OK, cool, so that is the official process. Thanks. Should I open a separate issue for this to be documented or is there already one?

**Comment by uncomfyhalomacro** [comment]
no movement

**Comment by razzeee** [comment]
I think this could be closed, as we allow to change appstream ids now, via an EOL
